e2e/teams.loggedin.spec.ts

```diff
@@ -0,0 +1,195 @@
+import { v4 as uuidv4 } from 'uuid'
+import { ADMIN_STORAGE_STATE, USER_STORAGE_STATE } from '../playwright.config'
+import { getLocalStorage } from './helpers/localStorage'
+import {
+  deleteTeamInvitationMessage,
+  deleteTeamInviteAcceptanceMessage,
+} from './helpers/messages'
+import {
+  USER_NAME_LOCALSTORAGE_KEY,
+  getAccessTokenFromCookie,
+  getAdminUserCredentials,
+  getUserIdFromLocalStorage,
+  goToDashboard,
+} from './helpers/testUser'
+
+const TEAM_NAME = 'swc-e2e-team-' + uuidv4()
+const INVITATION_MESSAGE = 'swc e2e test team invitation' + uuidv4()
+
+let adminPage: Page
+let userPage: Page
+
+test.describe('Teams', () => {
+  test.beforeAll(async ({ browser }) => {
+    adminPage = await browser.newPage({
+      storageState: ADMIN_STORAGE_STATE,
+    })
+    userPage = await browser.newPage({
+      storageState: USER_STORAGE_STATE,
+    })
+  })
+  test('should exercise team lifecycle', async ({}, testInfo) => {
+    const userName = await test.step('should get user name', async () => {
+      const userName = await getLocalStorage(
+        userPage,
+        USER_NAME_LOCALSTORAGE_KEY,
+      )
+      expect(userName).not.toBeNull()
+
+      return userName
+    })
+
+    const adminUserName = await test.step('should get admin name', async () => {
+      const { adminUserName } = getAdminUserCredentials()
+      return adminUserName
+    })
+
+    await test.step('user should create a team with a unique name', async () => {
+      await goToDashboard(userPage)
+      await userPage.getByLabel('Teams').click()
+      await expect(
+        userPage.getByRole('heading', { name: 'Your Teams' }),
+      ).toBeVisible()
+
+      await userPage.getByRole('button', { name: 'Create a New Team' }).click()
+      await userPage.getByRole('textbox').nth(1).fill(TEAM_NAME)
+      await userPage.getByRole('button', { name: 'OK' }).click()
+      await expect(
+        userPage.getByText(`Team Created: ${TEAM_NAME}`),
+      ).toBeVisible()
+    })
+
+    await test.step('user should invite admin to team', async () => {
+      await userPage.getByRole('button', { name: 'Team Actions' }).click()
+      await userPage.getByRole('link', { name: 'Invite User' }).click()
+
+      await userPage
+        .getByRole('textbox', { name: 'Enter a name or email address...' })
+        .type(adminUserName!)
+      await userPage
+        .getByRole('menuitem', { name: `@${adminUserName}` })
+        .click()
+      const loadInvitedUser = userPage.getByText('Loading...')
+      await expect(loadInvitedUser).toBeVisible()
+      await expect(loadInvitedUser).not.toBeVisible()
+
+      await userPage.getByLabel('Invitation Message').type(INVITATION_MESSAGE)
+
+      const spinner = userPage.locator('.modal-body > .spinner')
+      await Promise.all([
+        expect(spinner).toBeVisible(),
+        userPage.getByRole('button', { name: 'Send Invitation(s)' }).click(),
+      ])
+
+      await expect(spinner).not.toBeVisible()
```

Comment:
After the user clicks on the send invitation button, a spinner appears while waiting for some work to finish, then disappears when the work is done. The test expects the spinner to appear and disappear. However, depending on how quickly the work finishes, the spinner may appear and disappear before Playwright has checked for its existence. Use `Promise.all` to ensure that the check for the spinner is fired before the button is clicked and the spinner's appearance isn't missed. See more discussion [here](https://github.com/microsoft/playwright/issues/5470#issuecomment-1285640689) and an illustrated example [here](https://tally-b.medium.com/the-illustrated-guide-to-using-promise-all-in-playwright-tests-af7a98af3f32).

Reply:
I see--so you're basically priming the visibility check to start looking before the action is invoked. This is a useful pattern, and I think it might make sense to use in other contexts--one that comes to mind immediately is using mock-service-worker to update mocked server behavior when simulating a data mutation.